### PR TITLE
Add Noninteractive mode

### LINKED ISSUE #623: Neo-CLI non-interactive mode
How can I start CLI's non-interactive mode ?

### PR DESCRIPTION
**Pull request by Lichen9618**
Close #623 

config change part:
`   
 "NonInteractive": {
      "IsNonInteractive": false,
  },
`

**How to start neo-cli in noninteractive mode?**
1.  enable IsNonInteractive section, 

**Why we add this mode?**
This mode enable neo-cli can be deployed in some special environment, like k8s.

### CODE REVIEW

File: neo-cli/Shell/MainService.cs [hunk 1]
```diff
@@ -1259,6 +1265,48 @@ protected internal override void OnStart(string[] args)
                     password: Settings.Default.RPC.SslCertPassword,
                     maxGasInvoke: Settings.Default.RPC.MaxGasInvoke);
             }
+            if (useNonInteractive&&Settings.Default.NonInteractive.IsNonInteractive)
```
Comment (shargon) [inline neo-cli/Shell/MainService.cs:1268]: dotnet format

**Comment by Tommo-L** [comment]
@ProDog @cloud8little Could help have a test？

**Comment by ProDog** [comment]
> @ProDog @cloud8little Could help have a test？

OK,I will test.

**Comment by ProDog** [comment]
Test and found 2 problems:
1. For commands with uncertain parameter length, like `export key` and `claim gas`, it not work accurately;
2. Command cannot be executed if it after `show state` command.

All other commands work fine.

**Comment by ProDog** [comment]
Or non-interactive mode does not support these commands.

**Comment by Tommo-L** [comment]
> Or non-interactive mode does not support these commands.

Yes, we don't need to support these commands in non-interactive mode.

File: neo-cli/Settings.cs [hunk 2]
```diff
@@ -78,6 +80,19 @@ public RPCSettings(IConfigurationSection section)
         }
     }
 
+    internal class NonInteractiveSettings
+    {
+        public bool IsNonInteractive { get; }
+        public string CommandPath { get; }
```
Comment (shargon) [inline neo-cli/Settings.cs:86]: Where it's used `CommandPath`?

Comment (Lichen9618) [reply neo-cli/Settings.cs]: commands support has been removed, this attribute should be removed too.

**Comment by ProDog** [comment]
The default value of  `IsNonInteractive`  should be false?

**Comment by shargon** [comment]
Maybe it's easy to detect if it's interactive mode

**Review by Tommo-L — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Comment by Tommo-L** [comment]
Merge? Ankr are waiting for this.

**Review by shargon — APPROVED** [review]

**Comment by Tommo-L** [comment]
Ping?

**Pull request merged**